src/testTree.ts

```diff
@@ -99,15 +89,20 @@ export class TestTree extends DisposableBase {
   }
 
   private _update() {
-    for (const [workspaceFolder, workspaceRootItem] of this._rootItems) {
+    for (const workspaceFolder of this._vscode.workspace.workspaceFolders ?? []) {
       const rootSuite = new TeleSuite('', 'root');
-      for (const model of this._models.enabledModels().filter(m => m.config.workspaceFolder === workspaceFolder)) {
+      for (const model of this._models.enabledModels().filter(m => m.config.workspaceFolder === workspaceFolder.uri.fsPath)) {
         for (const project of model.enabledProjects())
           rootSuite.suites.push(project.suite as TeleSuite);
       }
-      const upstreamTree = new upstream.TestTree(workspaceFolder, rootSuite, [], undefined, path.sep);
+      const upstreamTree = new upstream.TestTree(workspaceFolder.uri.fsPath, rootSuite, [], undefined, path.sep);
       upstreamTree.sortAndPropagateStatus();
       upstreamTree.flattenForSingleProject();
+      if (upstreamTree.rootItem.children.length === 0) {
+        this._testController.items.delete(this._idWithGeneration(workspaceFolder.uri.fsPath));
```

Comment:
Does delete work fine if the element is not there?

Reply:
Yup!